sdk/core/azure-core/src/main/java/com/azure/core/util/logging/ClientLogger.java

```diff
@@ -151,20 +151,36 @@ private void log(int logLevel, String format, Object... args) {
     }
 
     /**
-     * This will log {@link RuntimeException}, if logging is enabled , and throw the runtime exception.
-     * @param runtimeException to be thrown. It will do nothing if {@code null} is provided.
-     * @throws RuntimeException which is requested by this call.
+     * Attempts to log the exception at the warning level and returns it to be thrown.
+     *
+     * @param runtimeException Runtime exception to be logged and returned.
+     * @return the passed {@code RuntimeException}
+     */
+    public RuntimeException logWarningAndThrow(RuntimeException runtimeException) {
```

Comment:
The name of this method is misleading. It doesn't 'throw' anymore.

Reply:
`logWarningToThrow`? `logExceptionAsWarning`? Any naming suggestions here?

Reply:
logExceptionAsWarning sounds better, as user needs to throw it manually.